packages/middleware-recursion-detection/src/index.spec.ts

```diff
@@ -70,6 +70,27 @@ describe(recursionDetectionMiddleware.name, () => {
     expect(request.headers[TRACE_ID_HEADER_NAME]).toBe("some-real-trace-id");
   });
 
+  it(`should NOT set ${TRACE_ID_HEADER_NAME} header when the header is already set in all lowercase`, async () => {
```

Comment:
"is already set in all lowercase" should be "is already set with some other casing"

add a couple of test cases with inconsistent casing like `x-AmZn-TrAcE-iD`